src/aws/flb_aws_util.c

```diff
@@ -796,6 +800,28 @@ flb_sds_t flb_get_s3_key(const char *format, time_t time, const char *tag, char
     s3_key = tmp_key;
     tmp_key = NULL;
 
+    /* Find all occurences of $INDEX and replace with the appropriate index. */
+    if (strstr((char *) format, INDEX_STRING)) {
+        seq_index_len = snprintf(NULL, 0, "%"PRIu64, seq_index);
+        seq_index_str = flb_malloc(seq_index_len + 1);
+        if (seq_index_str == NULL) {
+            goto error;
+        }
+
+        sprintf(seq_index_str, "%"PRIu64, seq_index);
+        seq_index_str[seq_index_len] = '\0';
+        tmp_key = replace_uri_tokens(s3_key, INDEX_STRING, seq_index_str);
+
+        if (strlen(tmp_key) > S3_KEY_SIZE) {
+            flb_warn("[s3_key] Object key length is longer than the 1024 character limit.");
+        }
+
+        flb_sds_destroy(s3_key);
+        s3_key = tmp_key;
+        tmp_key = NULL;
+        flb_free(seq_index_str);
+    }
+
     /* Find all occurences of $UUID and replace with a random string. */
```

Comment:
Do we need any `flb_free(seq_index_str )` in this function?

Reply:
If I understand correctly, we should only need one `flb_free` on line 822 because it is malloc-ed on line 806.